docs/classical_instructions.rst

```diff
@@ -1,4 +1,241 @@
+| "le"  |                               |                | "gt"          | k.classical(rd, operation(rs1, '<=', rs2)) |
++-------+                               +                +---------------+--------------------------------------------+
+| "ge"  |                               |                | "lt"          | k.classical(rd, operation(rs1, '>=', rs2)) |
++-------+                               +----------------+---------------+--------------------------------------------+
+| "and" |                               | BITWISE        |               | k.classical(rd, operation(rs1, '&', rs2))  |
++-------+                               +                +               +--------------------------------------------+
+| "or"  |                               |                |               | k.classical(rd, operation(rs1, '|', rs2))  |
++-------+                               +                +               +--------------------------------------------+
+| "xor" |                               |                |               | k.classical(rd, operation(rs1, '^', rs2))  |
++-------+-------------------------------+                +               +--------------------------------------------+
+| "not" | 1 dest and 1 src reg index    |                |               | k.classical(rd, operation('~', rs))        |
++-------+                               +----------------+               +--------------------------------------------+
+| "mov" |                               | ARITHMETIC     |               | k.classical(rd, operation(rs))             |
++-------+-------------------------------+                +               +--------------------------------------------+
+| "ldi" | 1 dest reg index, 1 imm_value |                |               | k.classical(rd, operation(3))              |
++-------+-------------------------------+----------------+               +--------------------------------------------+
+| "nop" | none                          | undefined      |               | k.classical('nop')                         |
++-------+-------------------------------+----------------+---------------+--------------------------------------------+
+
+In the above:
+
+``operation()`` creates an expression (binary, unary, register, or immediate); apart from in the OpenQL interface as shown above, it is also used as expression in the internal representation of the *br_condition* attribute of a kernel
+
+``operation type`` indicates the type of operation which is mainly used for checking
+
+``inv operation`` represents the inverse of the operation; it is used in code generation of conditional branching; see :ref:`kernel`
+
+
+Classical gates in circuits and bundles in the internal representation
+----------------------------------------------------------------------
+
+In circuits and bundles, no difference is made between classical and quantum gates.
+Classical gates are scheduled based on their operands and duration.
```

Comment:
Yes, they can be in parallel and therefore are scheduled together. What is your point?

Reply:
I guess I don't understand what scheduling of classical instructions really means. For both the CC-light and the CC, there is a single instruction memory with classical instructions and 'quantum' instructions interspersed. 
The quantum instructions are timed, and run in a different pipeline. Normally, the classical instructions run in parallel to the quantum instructions, and thus take no time. The CC even enforces that, it is an error for the 'quantum' pipeline to run empty, because in that case timing would be affected.
Apart from that, at least in the CC classical instructions run at a different rate (5 ns vs. 20 ns), and the duration is not fixed (i.e. due to register dependencies), so making accurate timing predictions is difficult.
Having said this: the CC backend currently does not support classical instructions, and in my opinion the whole OpenQL view op classical instructions should be revised (see issue #262)